projects/vdk-plugins/vdk-control-api-auth/README.md

```diff
@@ -6,7 +6,27 @@ utilities used by vdk-control-cli and other plugins.
 ## Usage
 
 This is a library plugin, not a runnable plugin, and it is intended to be
-used as a dependency for other plugins which need to authenticate users.
+used as a dependency for other plugins, which need to authenticate users.
+
+To use the library within a plugin or another Versatile Data Kit component,
+just import the `Authentication` class, and create an instance of it. The
+different authentication flows require different parameters to be specified.
+
+Once everything is done, in order to authenticate, call `.authenticate()` on
+the `Authentication` instance.
+
+Example Usage:
+```python
+from vdk.plugin.control_api_auth.authentication import Authentication
+
+auth = Authentication(
+    token="<oauth-api-token>",
+    authorization_url="https://some-authorization-endpoint",
+    auth_type="api-token",
+)
+
+auth.authenticate()
```

Comment:
How do I get the access token ?

Reply:
Good question. I've added a method to get the access token, though I wonder if we could just return the access token as part of the `.authenticate` call?

Reply:
It's better to be separate and explicit call (get oauth2 access token). As we'd have another type of authentication mechanism working at teh same time. As part of https://github.com/vmware/versatile-data-kit/pull/809/ I will be adding support for kebreros where authentication is using "Authentication: Negotiate" header. So probably I can add get_kerberos_auth_header method ...

Reply:
I see, ok.